homeassistant/components/alarm_control_panel/elkm1.py

```diff
@@ -0,0 +1,165 @@
+"""
+Each ElkM1 area will be created as a separate alarm_control_panel in HASS.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/alarm_control_panel.elkm1/
+"""
+
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+import homeassistant.components.alarm_control_panel as alarm
+from homeassistant.const import (ATTR_ENTITY_ID, STATE_ALARM_ARMED_AWAY,
+                                 STATE_ALARM_ARMED_HOME,
+                                 STATE_ALARM_ARMED_NIGHT, STATE_ALARM_ARMING,
+                                 STATE_ALARM_DISARMED, STATE_ALARM_PENDING,
+                                 STATE_ALARM_TRIGGERED, STATE_UNKNOWN)
+
+from homeassistant.components.elkm1 import (DOMAIN, create_elk_devices,
+                                            ElkDeviceBase,
+                                            register_elk_service)
+from elkm1_lib.const import AlarmState, ArmedStatus, ArmLevel, ArmUpState
```

Comment:
This does work. Is there a reason for importing only within functions?

Reply:
I understand the normal reasoning (that things would crash if trying to import a library not yet installed by HASS as a dependency). In this case it won't cause that problem (though it breaks with the established pattern) since trying to load the component by itself probably fails terribly.

The alarm_control_panel.elkm1 (and light and so on) never get loaded unless they were loaded by the top-level elkm1 component being configured, which means the library must be installed, which makes this safe (even if not "how it's done"). 

Even if we made things get imported only in functions, you couldn't load alarm_control_panel by itself without configuring the elkm1 platform.

If we have to move those imports into functions it means a lot of repetitive imports. If we have to, we have to, it's just annoying and ugly (and maybe impacts performance, but I imagine that after the first import the repeats are nearly zero cost).

Reply:
https://developers.home-assistant.io/docs/en/creating_platform_code_review.html#1-requirements

Reply:
So what is established is:
1. There's a rule.
2. The rule doesn't take into account when a platform derives from a base that has a requirement.
3. The Elk alarm_control_panel has no REQUIREMENTS because the REQUIREMENTS are in the base.
4. The Elk base `elkm1/__init__.py` follows the rule because the code won't run if the rule isn't followed (and, yah, I stumbled on this and it took an hour to figure out).

What would have to happen to improve the rule?

Reply:
The rule is there for a reason. We can't guard users from configuring a platform directly instead of the component. If the users would do that, there would be an error, since the platform is imported
before the dependencies are loaded and requirements installed.

Reply:
Thank-you. That helps.